containers/docker/deployment/admin/hpcc-tools/config_hpcc.sh

```diff
@@ -0,0 +1,428 @@
+         [ -z "$node_type2" ] &&  node_type2=$node_type
+         cluster_name=$(echo $node_type | cut -d'-' -s -f2);
+         [ -n "$cluster_name" ] && node_type2=${node_type2}-${cluster_name}
+         cp ${ipDir}/${node_type} ${ipDir2}/${node_type2}
+    done
+
+    [ "${found_dali}" = "true" ] && return  || :
+    [ -n "$cluster_node_types" ] && cluster_node_types="${cluster_node_types}," || :
+    cluster_node_types="${cluster_node_types}dali"
+    echo "cluster_node_types=${cluster_node_types}" > ${wkDir}/hpcc.conf
+
+    # Find dali ip
+    dali_ip=$($SUDOCMD /opt/HPCCSystems/sbin/configgen -env ${clusterConfigDir}/environment.xml -listall -t dali | cut -d',' -f3)
+    echo $dali_ip > ${ipDir2}/dali
+
+    if [ -e ${ipDir2}/node ]
+    then
+         node_to_process=node
+    else
+         node_to_process=support
+    fi
+    cat ${ipDir2}/${node_to_process} | grep -v ${dali_ip} > ${wkDir}/node_tmp || :
+
+    mv ${wkDir}/node_tmp ${ipDir2}/${node_to_process}
+    non_dali_nodes=$(cat ${ipDir2}/${node_to_process} | wc -l)
+    if [ ${non_dali_nodes} -eq 0 ]
+    then
+        rm -rf ${ipDir2}/${node_to_process}
+    fi
+    hpcc_config=$(ls ${ipDir2} | grep -v "admin" | tr '\n' ',')
+    echo "cluster_node_types=${hpcc_config%,}" > ${wkDir}/hpcc.conf
+}
+
+function setup_ansible_hosts()
+{
+  $SUDOCMD ${SCRIPT_DIR}/ansible/setup.sh -d ${ipDir2} -c ${wkDir}/hpcc.conf
+  export ANSIBLE_HOST_KEY_CHECKING=False
+}
+
+#------------------------------------------
+# Need root or sudo
+#
+SUDOCMD=
+[ $(id -u) -ne 0 ] && SUDOCMD=sudo
+
+
+#------------------------------------------
+# LOG
+
+#
+LOG_DIR=~/tmp/log/hpcc-tools
+mkdir -p $LOG_DIR
+LONG_DATE=$(date "+%Y-%m-%d_%H-%M-%S")
+LOG_FILE=${LOG_DIR}/config_hpcc_${LONG_DATE}.log
+touch ${LOG_FILE}
+exec 2>$LOG_FILE
+set -x
+
+update=0
+appName=hpcc
+ipDir=~/tmp/ips
+ipDir2=~/tmp/ips2
+wkDir=~/tmp/work
+notPush=0
+notGetIps=0
+notCreateEnv=0
+HPCC_HOME=/opt/HPCCSystems
+clusterConfigDir=/etc/HPCCSystems/source
+numSupport=
+numEsp=
+numRoxie=
+numThor=
+networkName=${appName}_ovnet
+
+[ ! -d ${wkDir} ] && mkdir -p ${wkDir}
+
+while getopts "*a:d:D:e:hn:N:r:s:t:uxX" arg
+do
+   case $arg in
+      a) appName=${OPTARG}
+         ;;
+      d) ipDir=${OPTARG}
+         ;;
+      D) HPCC_HOME=${OPTARG}
+         ;;
+      e) numEsp=${OPTARG}
+         ;;
+      h) usage
+         ;;
+      n) networkName=${OPTARG}
+        ;;
+      N) notPush=1
+        ;;
+      r) numRoxie=${OPTARG}
+         ;;
+      s) numSupport=${OPTARG}
+         ;;
+      t) numThor=${OPTARG}
+         ;;
+      u) update=1
+        ;;
+      x) notGetIps=1
+        ;;
+      X) notCreateEnv=1
+        ;;
+      ?)
+         echo "Unknown option $OPTARG"
+         usage
+         ;;
+   esac
+done
+
+echo "update mode: $update"
+#----------------------------------------o
+# Start sshd
+#
+ps -efa | grep -v sshd |  grep -q sshd ||  $SUDOCMD mkdir -p /var/run/sshd; $SUDOCMD  /usr/sbin/sshd -D &
```

Comment:
This might not be an issue, but how do we ensure these commands will be available on the target OS?

Reply:
Yes, the commands are available for the base Docker images we select.